### DISCUSSION TITLE
Why eventLoopGroup that came from Hyperfoil must be set to the HotRod

### DISCUSSION THREAD

**Discussion by diegolovison** [Hyperfoil/Hyperfoil#473] [Q&A]
Why the `eventLoopGroup` that came from Hyperfoil must be set to the HotRod?

https://github.com/Hyperfoil/Hyperfoil/blob/master/hotrod/src/main/java/io/hyperfoil/hotrod/connection/HotRodRemoteCache
PoolImpl.java#L58

**Comment by diegolovison** [comment]
cc @rvansa

**Reply by rvansa** [reply]
The architecture of Hyperfoil is intentionally single-thread-per-session, and we're cautious to not introduce any synchronization between event loop threads. Without this you have no control on which thread the reply will come.

**Comment by franz1981** [comment]
Thanks @rvansa - and the other reason is because we want to keep the write/send request's path "straight" i.e. 
- Netty, if the caller thread is the event loop which own the channel can perform writing to the socket inline (this is happening under the hood on `flush` operations)
- conversely, if the caller thread is not right, it dispatch the request via an ad-hoc task - adding an indirection layer

**Comment by diegolovison** [comment]
Thanks all!
